src/traces/scattergl/index.js

```diff
@@ -47,116 +48,73 @@ function calc(gd, trace) {
     var xa = Axes.getFromId(gd, trace.xaxis);
     var ya = Axes.getFromId(gd, trace.yaxis);
     var subplot = fullLayout._plots[trace.xaxis + trace.yaxis];
+    var count = trace._length;
+    var count2 = count * 2;
     var stash = {};
+    var i, xx, yy;
 
+    var x = xa.makeCalcdata(trace, 'x');
+    var y = ya.makeCalcdata(trace, 'y');
 
-    var x = xaxis.type === 'linear' ? trace.x : xaxis.makeCalcdata(trace, 'x');
-    var y = yaxis.type === 'linear' ? trace.y : yaxis.makeCalcdata(trace, 'y');
-
-    var count = trace._length, i, xx, yy;
+    // we need hi-precision for scatter2d,
+    // regl-scatter2d uses NaNs for bad/missing values
+    //
+    // TODO should this be a Float32Array ??
+    var positions = new Array(count2);
+    for(i = 0; i < count; i++) {
+        xx = x[i];
+        yy = y[i];
+        // TODO does d2c output any other bad value as BADNUM ever?
+        positions[i * 2] = xx === BADNUM ? NaN : xx;
+        positions[i * 2 + 1] = yy === BADNUM ? NaN : yy;
+    }
 
-    if(!x) {
-        x = Array(count);
-        for(i = 0; i < count; i++) {
-            x[i] = i;
+    if(xa.type === 'log') {
+        for(i = 0; i < count2; i += 2) {
+            positions[i] = xa.d2l(positions[i]);
```

Comment:
🐎 we've already been through `makeCalcData` so we should be able to do `c2l` here

Reply:
Thanks for pointing this out! Done in -> 
98d2407